gnucash/register/register-gnome/gnucash-sheet.c

```diff
@@ -1465,9 +1465,7 @@ gnucash_sheet_button_press_event (GtkWidget *widget, GdkEventButton *event)
         button_1 = TRUE;
         break;
     case 2:
-        if (event->type != GDK_BUTTON_PRESS)
-            return FALSE;
-        gnc_item_edit_paste_clipboard (GNC_ITEM_EDIT(sheet->item_editor));
+        // This middle button option has been moved to gnucash-item-edit.c
```

Comment:
It may be a bit more complicated than this.
The intent of a middle mouse button click is to paste the current clipboard at the current location.
By no longer handling a middle mouse button click here this works when clicking in the item_edit (we get this for free from Gtk).

However a middle-click somewhere else on the sheet now does nothing anymore. I think it should
- move the cursor to where the user clicked (like a left-click)
- pass the middle-click on to the item-edit to handle the past action

Note that if you get to redo the itemedit popup as I suggested in a previous PR, this function really reduces to moving the cursor and passing any event on to the itemedit in its new location...

Reply:
This would be nice but it does not happen in 2.6 version. What I was aiming at getting was what is missing working and then do the focus change on master as I do not think it will be that straightforward for me.

Reply:
Hmm, not sure I can follow. 2.6 is not a reference as it was still using gnomecanvas. I already made plenty of changes for 3.0 in this area. Pasting already works in the itemedit. What's not working is moving the cursor on a middle click. IMO that can probably solved by handling left-click and middle-click exactly the same here in the sheet. You're now returning prematurely for a middle-click.

When I said it may be a bit more complicated I was thinking of how to process the event such that it gets passed to the itemedit after the cursor is updated. But we can wait with that part if you prefer.